### ISSUE TITLE
AnnotationOptions.alpha Needs To Return SuperType

### ISSUE THREAD

**Issue by bleege** [mapbox/mapbox-gl-native#2631]
Currently `AnnotationOptions.alpha()` [returns an `AnnotationsOptions` object](https://github.com/mapbox/mapbox-gl-native/blob/074d61a55e32e83173ec12b2eb34b78ddaaf0ef1/android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/AnnotationOptions.java#L9).  While this is functional, it requires casting to be applied in common Builder patterns which isn't good.  For example:

```java
MapView map = ...;

// works
map.addPolygon(new PolygonOptions().add(latLngs));

// currently doesn't work due to typing, but is ideal
map.addPolygon(new PolygonOptions().add(latLngs).alpha(0.5f));

// casting needed to make work, but not something that should be used 
map.addPolygon((PolygonOptions) new PolygonOptions().add(latLngs).alpha(0.5f));
```

/cc @tobrun @ljbade

**Comment by ljbade** [comment]
Interesting, the other options support this. Also core GL marker don't support alpha, so only needed for lines and polygons.

**Comment by tobrun** [comment]
@bleege didn't the generic builder pattern solve this?

**Comment by bleege** [comment]
@tobrun I didn't have time to finish yesterday afternoon so I just created this issue to make sure this was known.  Hoping to get it to today.

**Comment by bleege** [comment]
Looking at Google's implementation they avoided this problem all together by keeping [PolygonOptions](https://developers.google.com/android/reference/com/google/android/gms/maps/model/PolygonOptions), [PolyLineOptions](https://developers.google.com/android/reference/com/google/android/gms/maps/model/PolylineOptions), and [MarkerOptions](https://developers.google.com/android/reference/com/google/android/gms/maps/model/MarkerOptions) all independent.  In other words, they don't use [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself) and each one of these Option classes extends from `java.lang.Object` and implements `android.os.Parcelable`.

**Comment by bleege** [comment]
Doing some R&D on this shows that the `AnnotationsOptions` abstract class came as part of clean up / refactoring in #1716.  In other words this [was introduced in Mapbox](https://github.com/hallahan/mapbox-gl-native/commit/9e3f60756f6e70fa8f4b0d5827127bd404b1baf6) and not needed as part of the Google API (confirms previous statement on API).

**Comment by ljbade** [comment]
@bleege Does the same apply to the `Marker`, `Polyline`, `Polygon` classes? If we split these up we will need to do a bit of re-factoring in `MapView`.

Also should point out this was an easy fix in my large PR, I just pushed `.alpha` to the concrete classes.

**Comment by bleege** [comment]
I decided to remove `AnnotationsOptions` and `MultiPointOptions` as they were for abstract classes that are not part of the Google Maps API.  I moved their features to concrete implementations which mirrors Google and also solves this issue.  Tests (see below) look good.  I also added an explicit usage of `.alpha()` to the Polygon Annotation.  Was able to confirm that the builder pattern worked code wise and more importantly the polygon actually was rendered using the proper alpha.

![device-2015-10-16-174451](https://cloud.githubusercontent.com/assets/989928/10555094/ae57a9ac-742f-11e5-8527-bdc3a59fe739.png)
_Test, Test, Test_

![device-2015-10-16-174638](https://cloud.githubusercontent.com/assets/989928/10555099/b7f679ac-742f-11e5-9b37-f4f8c398c2af.png)
_Original Polygon, No Alpha_

![device-2015-10-16-174516](https://cloud.githubusercontent.com/assets/989928/10555100/c08ec858-742f-11e5-8e4e-0ab01b89a8f3.png)
_Original Polygon, 0.5 Alpha_
